src/AppInstallerCLICore/Workflows/WorkflowBase.h

```diff
@@ -231,13 +244,13 @@ namespace AppInstaller::CLI::Workflow
     // Outputs: Package
     struct EnsureOneMatchFromSearchResult : public WorkflowTask
     {
-        EnsureOneMatchFromSearchResult(bool isFromInstalledSource) :
-            WorkflowTask("EnsureOneMatchFromSearchResult"), m_isFromInstalledSource(isFromInstalledSource) {}
+        EnsureOneMatchFromSearchResult(SearchPurpose purpose) :
+            WorkflowTask("EnsureOneMatchFromSearchResult"), m_searchPurpose(purpose) {}

```

Comment:
Check expected searchPurpose upfront unless this function can handle all of them

Reply:
This can handle all of them, and defaults to `Resource::String::NoPackageFound` in the event new enum values are added